Mage.Sets/src/mage/cards/s/ShanidSleepersScourge.java

```diff
@@ -0,0 +1,115 @@
+package mage.cards.s;
+
+import mage.MageInt;
+import mage.abilities.TriggeredAbilityImpl;
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.effects.common.DrawCardSourceControllerEffect;
+import mage.abilities.effects.common.LoseLifeSourceControllerEffect;
+import mage.abilities.effects.common.continuous.GainAbilityControlledEffect;
+import mage.abilities.keyword.MenaceAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.*;
+import mage.filter.common.FilterCreaturePermanent;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.permanent.Permanent;
+import mage.game.stack.Spell;
+import mage.players.Player;
+
+import java.util.UUID;
+
+/**
+ * @author freaisdead
+ */
+public final class ShanidSleepersScourge extends CardImpl {
+    private static final FilterCreaturePermanent otherLegendaryCreaturesFilter = new FilterCreaturePermanent();
+
+    static {
+        otherLegendaryCreaturesFilter.add(SuperType.LEGENDARY.getPredicate());
+    }
+
+    public ShanidSleepersScourge(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{1}{R}{W}{B}");
+
+        this.addSuperType(SuperType.LEGENDARY);
+        this.subtype.add(SubType.HUMAN);
+        this.subtype.add(SubType.KNIGHT);
+        this.power = new MageInt(2);
+        this.toughness = new MageInt(4);
+
+        // Menace
+        this.addAbility(new MenaceAbility());
+
+        // Other legendary creatures you control have menace.
+        this.addAbility(new SimpleStaticAbility(new GainAbilityControlledEffect(
+                new MenaceAbility(false),
+                Duration.WhileOnBattlefield,
+                otherLegendaryCreaturesFilter,
+                true)));
+        // Whenever you play a legendary land or cast a legendary spell, you draw a card and you lose 1 life.
+        this.addAbility(new DrawAndLoseLife(1, 1));
+    }
+
+    private ShanidSleepersScourge(final ShanidSleepersScourge card) {
+        super(card);
+    }
+
+    @Override
+    public ShanidSleepersScourge copy() {
+        return new ShanidSleepersScourge(this);
+    }
+}
+
+
+class DrawAndLoseLife extends TriggeredAbilityImpl {
+
+    public DrawAndLoseLife(int drawAmount, int loseLifeAmount) {
+        super(Zone.BATTLEFIELD, new LoseLifeSourceControllerEffect(loseLifeAmount), false);
+        this.addEffect(new DrawCardSourceControllerEffect(drawAmount));
+    }
+
+    public DrawAndLoseLife(DrawAndLoseLife ability) {
+        super(ability);
+    }
+
+    @Override
+    public boolean checkEventType(GameEvent event, Game game) {
+        return event.getType() == GameEvent.EventType.LAND_PLAYED || event.getType() == GameEvent.EventType.SPELL_CAST;
+    }
+
+    @Override
+    public boolean checkTrigger(GameEvent event, Game game) {
+        if (event.getPlayerId().equals(this.getControllerId())) {
```

Comment:
The reason it's probably not triggering is probably this check. It looks like you're missing a `!`, it *should* trigger if the player who caused the event is the ability's controller.

Reply:
can we have a quick call with Discord (i'll send you in gitter my account) ? because I'm not getting the way this should be implemented @Alex-Vasile